modules/azure/availabilityset.go

```diff
@@ -0,0 +1,132 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// AvailabilitySetExists indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExists(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) bool {
+	exists, err := AvailabilitySetExistsE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// AvailabilitySetExistsE indicates whether the speficied Azure Availability Set exists
+func AvailabilitySetExistsE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetAvailabilitySetE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// CheckAvailabilitySetContainsVM checks if the Virtual Machine is contained in the Availability Set VMs
+func CheckAvailabilitySetContainsVM(t testing.TestingT, vmName string, avsName string, resGroupName string, subscriptionID string) bool {
+	avsVMs, err := GetAvailabilitySetVMsE(t, avsName, resGroupName, subscriptionID)
+	if err != nil {
+		return false
+	}
+
+	return collections.ListContains(avsVMs, strings.ToLower(vmName))
+}
+
+// GetAvailabilitySetVMs gets a list of VM names in the specified Azure Availability Set
+func GetAvailabilitySetVMs(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) []string {
+	vms, err := GetAvailabilitySetVMsE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return vms
+}
+
+// GetAvailabilitySetVMsE gets a list of VM names in the specified Azure Availability Set
+func GetAvailabilitySetVMsE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) ([]string, error) {
+	client, err := GetAvailabilitySetClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	avs, err := client.Get(context.Background(), resGroupName, avsName)
+	if err != nil {
+		return nil, err
+	}
+
+	vms := []string{}
+
+	for _, vm := range *avs.VirtualMachines {
+		vms = append(vms, strings.ToLower(GetNameFromResourceID(*vm.ID)))
+	}
+
+	return vms, nil
+}
+
+// GetAvailabilitySetFaultDomainCount gets the Fault Domain Count for the specified Azure Availability Set
+func GetAvailabilitySetFaultDomainCount(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) int32 {
+	avsFaultDomainCount, err := GetAvailabilitySetFaultDomainCountE(t, avsName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return avsFaultDomainCount
+}
+
+// GetAvailabilitySetFaultDomainCountE gets the Fault Domain Count for the specified Azure Availability Set
+func GetAvailabilitySetFaultDomainCountE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (int32, error) {
```

Comment:
Sanity check: this is a very thing wrapper around the underlying Azure SDK. In our [contribution guidelines](https://terratest.gruntwork.io/docs/community/contributing/#infrastructure-and-validation-helpers), to avoid Terratest becoming too bloated, we typically prefer to only include helpers that wrap some degree of complexity. Of course, this is balanced by the need to keep code short/DRY for very common operations, so if you expect a lot of test code to be looking up the fault domain count, then a short method for doing that may be worth maintaining.

Reply:
Kept since is helpful. Reduced complexity.

Reply:
The method was shortened, also kept as it provides a useful property return and a sample for similar getters.